content/300-guides/100-performance-and-optimization/090-metrics.mdx

```diff
@@ -0,0 +1,493 @@
+pool_idle_connections 21
+
+# HELP pool_wait_count The number of workers waiting for a connection.
+# TYPE pool_wait_count gauge
+pool_wait_count 0
+
+# HELP query_active_transactions
+# TYPE query_active_transactions gauge
+query_active_transactions 0
+
+# HELP pool_wait_duration_ms The wait time for a worker to get a connection.
+# TYPE pool_wait_duration_ms histogram
+pool_wait_duration_ms_bucket{le="0"} 0
+pool_wait_duration_ms_bucket{le="1"} 3
+pool_wait_duration_ms_bucket{le="5"} 3
+pool_wait_duration_ms_bucket{le="10"} 3
+pool_wait_duration_ms_bucket{le="50"} 3
+pool_wait_duration_ms_bucket{le="100"} 3
+pool_wait_duration_ms_bucket{le="500"} 3
+pool_wait_duration_ms_bucket{le="1000"} 3
+pool_wait_duration_ms_bucket{le="5000"} 3
+pool_wait_duration_ms_bucket{le="50000"} 3
+pool_wait_duration_ms_bucket{le="+Inf"} 3
+pool_wait_duration_ms_sum 0.023208
+pool_wait_duration_ms_count 3
+
+# HELP query_operation_total_elapsed_time_ms
+# TYPE query_operation_total_elapsed_time_ms histogram
+query_operation_total_elapsed_time_ms_bucket{le="0"} 0
+query_operation_total_elapsed_time_ms_bucket{le="1"} 1
+query_operation_total_elapsed_time_ms_bucket{le="5"} 2
+query_operation_total_elapsed_time_ms_bucket{le="10"} 2
+query_operation_total_elapsed_time_ms_bucket{le="50"} 2
+query_operation_total_elapsed_time_ms_bucket{le="100"} 2
+query_operation_total_elapsed_time_ms_bucket{le="500"} 2
+query_operation_total_elapsed_time_ms_bucket{le="1000"} 2
+query_operation_total_elapsed_time_ms_bucket{le="5000"} 2
+query_operation_total_elapsed_time_ms_bucket{le="50000"} 2
+query_operation_total_elapsed_time_ms_bucket{le="+Inf"} 2
+query_operation_total_elapsed_time_ms_sum 3.197624
+query_operation_total_elapsed_time_ms_count 2
+
+# HELP query_total_elapsed_time_ms
+# TYPE query_total_elapsed_time_ms histogram
+query_total_elapsed_time_ms_bucket{le="0"} 0
+query_total_elapsed_time_ms_bucket{le="1"} 5
+query_total_elapsed_time_ms_bucket{le="5"} 5
+query_total_elapsed_time_ms_bucket{le="10"} 5
+query_total_elapsed_time_ms_bucket{le="50"} 5
+query_total_elapsed_time_ms_bucket{le="100"} 5
+query_total_elapsed_time_ms_bucket{le="500"} 5
+query_total_elapsed_time_ms_bucket{le="1000"} 5
+query_total_elapsed_time_ms_bucket{le="5000"} 5
+query_total_elapsed_time_ms_bucket{le="50000"} 5
+query_total_elapsed_time_ms_bucket{le="+Inf"} 5
+query_total_elapsed_time_ms_sum 1.8407059999999997
+query_total_elapsed_time_ms_count 5
+```
+
+### Use Prisma metrics with the Prometheus metrics system
+
+In the majority of cases, Prometheus must scrape an endpoint to retrieve metrics. The following example shows how to send data with `Express.js`:
+
+<!-- prettier-ignore -->
+```js
+import { PrismaClient } from "@prisma/client";
+import express, { Request, Response } from "express";
+
+const app = express();
+const port = 4000;
+const prisma = new PrismaClient();
+
+app.get("/metrics", async (_req, res: Response) => {
+  let metrics = await prisma.$metrics.prometheus();
+  res.end(metrics);
```

Comment:
Should this be `res.end` or `res.send`?

Reply:
I think you're right

Reply:
According to @SevInf , res.end will work here and is more lightweight, so I'll leave it as-is.